crates/oxc_linter/src/rules/unicorn/no_unreadable_array_destructuring.rs

```diff
@@ -0,0 +1,148 @@
+use oxc_allocator::Vec;
+use oxc_ast::{
+    ast::{AssignmentTarget, AssignmentTargetPattern},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-unicorn(no-unreadable-array-destructuring): Disallow unreadable array destructuring")]
+#[diagnostic(
+    severity(warning),
+    help("Array destructuring may not contain consecutive ignored values.")
+)]
+struct NoUnreadableArrayDestructuringDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoUnreadableArrayDestructuring;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow unreadable array destructuring
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Destructuring is very useful, but it can also make some code harder to read.
+    /// This rule prevents ignoring consecutive values when destructuring from an array.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// const [,, foo] = parts;
+    ///
+    /// // Good
+    /// const [foo] = parts;
+    /// ```
+    NoUnreadableArrayDestructuring,
+    style
+);
+
+fn is_unreadable_array_destructuring<T, U>(elements: &Vec<Option<T>>, rest: &Option<U>) -> bool {
+    if elements.len() >= 3
+        && elements
+            .iter()
+            .enumerate()
+            .any(|(index, el)| el.is_none() && elements[index + 1].is_none())
```

Comment:
```suggestion
  &&  elements.windows(2).any(|window| window.iter().all(Option::is_none))
  ```
  
  i think this may panic otherwise

Reply:
Thank you for correction.

Reply:
looks like [IWANABETHATGUY](https://github.com/IWANABETHATGUY) got there first 🙂